src/script/lua_api/l_metadata.cpp

```diff
@@ -172,6 +173,13 @@ int MetaDataRef::l_get_float(lua_State *L)
 	return 1;
 }
 
+static std::string double_to_string(f64 number) {
+	// Do not use Lua to convert number to string, since that may lose precision.
+	char buf[64];
+	snprintf(buf, 64, "%.17g", number);
+	return buf;
+}
```

Comment:
```suggestion
static inline std::string double_to_string(f64 number)
{
	// Do not use Lua to convert number to string, since that may lose precision.
	char buf[64];
	snprintf(buf, sizeof(buf), "%.17g", number);
	return buf;
}
```